integration_tests/integration_tests/dbt_project.py

```diff
@@ -0,0 +1,125 @@
+
+        self.sql_parser = SqlBlockParser(self.config, self.manifest, self.config)
+
+        self.relations_to_cleanup = []
+
+    def execute_macro(self, macro_name, **kwargs):
+        if "." in macro_name:
+            package_name, actual_macro_name = macro_name.split(".", 1)
+        else:
+            package_name = None
+            actual_macro_name = macro_name
+
+        return self.adapter.execute_macro(
+            macro_name=actual_macro_name,
+            project=package_name,
+            kwargs=kwargs,
+            manifest=self.manifest
+        )
+
+    def execute_sql(self, sql: str):
+        temp_node = self._create_temp_node(sql)
+
+        try:
+            sql_compiler = SqlCompileRunner(
+                self.config, self.adapter, node=temp_node, node_index=1, num_nodes=1
+            )
+            compiled_node = sql_compiler.compile(self.manifest)
+            compiled_sql = getattr(compiled_node, COMPILED_CODE)
+            return self.adapter.execute(compiled_sql, fetch=True)[1]
+        finally:
+            self.clear_node(temp_node.name)
+
+    def create_relation(self, database: Optional[str], schema: Optional[str], name: str,
+                        relation_type: str = "table") -> BaseRelation:
+        return self.adapter.Relation.create(database, schema, name, type=relation_type)
+
+    def create_table_as(self, relation, sql, temporary):
+        if temporary and self.adapter_name in ["spark", "databricks"]:
+            temporary = False
+            self.relations_to_cleanup.append(relation)
+
+        create_table_kwargs = {
+            "temporary": temporary,
+            "relation": relation
+        }
+        if dbt_version >= version.parse("1.3.0"):
+            create_table_kwargs["compiled_code"] = sql
+        else:
+            create_table_kwargs["sql"] = sql
+
+        create_table_query = self.execute_macro("dbt.create_table_as", **create_table_kwargs)
+        self.execute_sql(create_table_query)
+
+    def cleanup(self):
+        for relation in self.relations_to_cleanup:
+            print("Dropping relation: %s" % relation)
+            self.execute_macro("dbt.drop_relation_if_exists", relation=relation)
+
+    def _create_temp_node(self, sql: str):
+        """Get a node for SQL execution against adapter"""
+        temp_node_name = str(uuid.uuid4())
+        self.clear_node(temp_node_name)
+        sql_node = self.sql_parser.parse_remote(sql, temp_node_name)
+        process_node(self.config, self.manifest, sql_node)
+        return sql_node
+
+    def clear_node(self, name: str):
+        """Removes the statically named node created by `execute_sql` and `compile_sql` in `dbt.lib`"""
+        self.manifest.nodes.pop(f"{NodeType.SqlOperation}.{self.config.project_name}.{name}", None)
+
+
+class Args(BaseModel):
+    """
+    Minimal mock to dbt config arguments
+    """
+    project_dir: str
+    target: Optional[str] = None
+    threads: Optional[int] = 1
```

Comment:
Maybe set default `threads` to 8 / 32?

Reply:
I actually did it as a safety because I saw a potential issue with accessing the adapter I create here in multiple threads, but not 100% sure about it.
In any case though I think it doesn't really impact anything - because everything I'm invoking here is synchronous.